pkg/certificate/providers/config.go

```diff
@@ -218,47 +230,47 @@ func (c *Config) getTresorOSMCertificateManager() (certificate.Manager, debugger
 	return certManager, certManager, nil
 }
 
-// GetCertFromKubernetes returns a Certificater type corresponding to the root certificate.
+// GetCertFromKubernetes is a helper function that loads a certificate from a Kubernetes secret
 // The function returns an error only if a secret is found with invalid data.
-func (c *Config) GetCertFromKubernetes() (certificate.Certificater, error) {
-	rootCertSecret, err := c.kubeClient.CoreV1().Secrets(c.providerNamespace).Get(context.Background(), c.caBundleSecretName, metav1.GetOptions{})
+func GetCertFromKubernetes(ns string, secretName string, kubeClient kubernetes.Interface) (certificate.Certificater, error) {
```

Comment:
I'd rename this to `GetRootCertFromKubernetes` or even better - `GetCABundleFromKubernentes`

Reply:
As I pointed above, not going to be exclusive for root/ca bundle anymore (hence the generalization).